### PR TITLE
Fix create user button on home page

### LINKED ISSUE #201: Add a User Link doesn't work
**Describe the bug**
As an admin, I want to be able to add a user.  When  I click the button I don't get a form to add a user, I get the flash message that I'm already logged in.  Currently one could log out to create a new account.  However, for our users they will have a volunteer or admin logged in checking out carriers etc, and we need them to be able to add users as well.  

**To Reproduce**
1. Go to `/home`
2. Click on `Add a user`

**Expected behavior**
I should get a new page to add another user.

### PR DESCRIPTION
**Pull request by sarslanoglu**
Resolves #201 

### Description

Create User button on home page is not working. With this addition admin can add users.

### Type of change

* Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

Manually by visiting pages

### CODE REVIEW

**Comment by sarslanoglu** [comment]
I'm not sure this is the desired solution but right know admin user can create another user by using same signup page. But in the end it logout as admin and sign in as newly created user. If need i can take the longer part and add new custom way for adding new users

**Comment by pollygee** [comment]
Ah Ok I didn't realize this wasn't already functioning, I thought it got broken.

An admin or a volunteer should be able to click "add a user" and create an account, same info as what exists now.  But it should not log out the volunteer or admin.  Eventually we will want to add functionality that upon creating the user, we will go to the new users agreements so they can sign the agreements.  At a meeting, volunteers will sign people up and then lend them items so they should remain logged in for all of this.  

If you're able to tackle this, it would be great thanks so much!

File: app/controllers/users/registrations_controller.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 module Users
   class RegistrationsController < Devise::RegistrationsController
+    skip_before_action :require_no_authentication, only: [:new, :create]
```
Comment (sarslanoglu) [inline app/controllers/users/registrations_controller.rb:5]: This required for logged in user to create other users

File: app/views/users/registrations/_form_fields.html.erb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 <div class="field">
   <%= f.label :first_name %><br />
   <%= f.text_field :first_name, required: true %>
+</div>
 
-</div><div class="field">
+<div class="field">
```
Comment (sarslanoglu) [inline app/views/users/registrations/_form_fields.html.erb:6]: Quick fix for wrong indentation

File: config/routes.rb [hunk 3]
```diff
@@ -6,7 +6,12 @@
 
   get 'home/index'
 
-  resources :users, only: [:index]
+  as :admin do
+    get  'volunteers',  to: 'users#new'
+    post 'volunteers',  to: 'users#add_user'
+  end
```
Comment (sarslanoglu) [inline config/routes.rb:12]: Here 'volunteers' is arbitrary. It can be anything but not user or users, because devise gem is reserving these for itself

**Comment by sarslanoglu** [comment]
@pollygee This time admin can add users but it has to fill all form values because there is validation for user. I think this version is what you desired. Fell free to switch branch and try. If pr gets ok, i will also fix code climate issues

**Comment by pollygee** [comment]
Hi @sarslanoglu  thanks so much for your work on this.  I just pulled your branch and when I create a user it signs me in as that user and that's not what we want. 

At a meeting a volunteer or admin will be signed into a computer or tablet, and we want them to be able to check out carriers to people as well as sign people up to become members.  This will mean that as they are signed in they need to be able to have someone sign up (while they're signed in) but the admin or volunteer needs to stay logged in, we don't want them logged out.

I think we should take out the auto login feature.  So if you do create a login without being logged in you'll just have to enter username and pw, but if an admin or volunteer is signed in they stay signed in.  Do you think you could add that?  @abronkema FYI

**Comment by sarslanoglu** [comment]
@pollygee That's strange. My branch right now does exactly what you want/tell above. But it gives different behaviour on your computer. I recorded a 1 minute video. Could you check it?

[user_save.mov.zip](https://github.com/rubyforgood/babywearing/files/3825770/user_save.mov.zip)

File: app/views/layouts/_header.html.erb [hunk 4]
```diff
@@ -8,11 +8,15 @@
 
     <div class="collapse navbar-collapse" id="navbarSupportedContent">
       <ul class="navbar-nav mr-auto">
-        <li class="nav-item active background-blue">
-          <%= link_to new_user_registration_path, class: 'nav-link nav-icon nav-first' do %>
-            <%= image_tag "BW- CREATE USER ICON.svg", alt:"Create a User" %><p>CREATE USER</p>
+        <% if user_signed_in? %>
```
Comment (pollygee) [inline app/views/layouts/_header.html.erb:11]: Does this mean that any user can add another user?  I think we only want volunteers and admins to be able to add new users.

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,65 @@
+<% if user_signed_in? %>
+  <% if user_can_add_new_member(current_user) %>
+	<h1>Create User</h1>
+
+	<%= form_for User.new , :url => {:action => "add_user"} do |form| %>
```
Comment (pollygee) [inline app/views/users/new.html.erb:5]: Do we have to recreate this form?  We can't use the current form we have?

**Comment by pollygee** [comment]
@sarslanoglu that is strange.  Your video does show the wanted behavior. I'm not sure why it's not showing the same on my machine.

File: app/views/layouts/_header.html.erb (hunk 4 above)
Comment (sarslanoglu) [reply app/views/layouts/_header.html.erb:11]: At first I only checked that `<% if user_can_add_new_member(current_user) %>` But then if there is no currently sign in user present, it got error. So basically first I check sign in user than control user_can_add_new_member before show Create User button

File: app/views/users/new.html.erb (hunk 5 above)
Comment (sarslanoglu) [reply app/views/users/new.html.erb:5]: I can use previous one it won't be a problem for me. Do you want me to just redirect sign up page or add some custom text and buttons with it?

**Comment by dixpac** [comment]
@sarslanoglu thanks for working this out 💛 .
I'm gonna close this for now because PR #159 is already addressing controller actions to `create and update` the user. We were waiting for #159 so we can link correct route on the navigation.

When 159 is merged I can ping you so you can link `Create user` to the correct route which is going to be `new_user_url..`, if you are up for it :) ?

**Comment by dixpac** [comment]
Whoops...I've just merged #159 :)...You can now open new PR or merge/rebase from master and add you fix on this one.

I would love if we could handle "manual" user creation through `users#create`. Only hassle is gonna be for that user to log in he/she will need to set password on the initial login

**Comment by sarslanoglu** [comment]
@dixpac Sure I can change my pr and open a new one using `users#create`

**Pull request closed without merging**